Sources/Event.swift

```diff
@@ -848,14 +850,43 @@ extension Signal.Event {
 			return { event in
 				switch event {
 				case let .value(value):
+					state.modify { state in
+						state.pendingValue = value
+					}
 					let date = scheduler.currentDate.addingTimeInterval(interval)
 					d.inner = scheduler.schedule(after: date) {
-						action(.value(value))
+						state.modify { state in
+							defer {
+								if state.pendingValue != nil {
+									state.pendingValue = nil
+									state.previousDate = date
+								}
+							}
+							if let pendingValue = state.pendingValue {
+								action(.value(pendingValue))
+							}
+						}
 					}
 
 				case .completed, .failed, .interrupted:
```

Comment:
Interruptions are generally immediate. I think introducing a delay here might be a problem. (I also think it's generally unexpected that interrupting would send another value. If you've canceled the work, then you're not interested in any other results.)

@andersio?

Reply:
I agree, and `failed` should follow suit since by contract [`failures` behave like exceptions and propagate immediately](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Documentation/APIContracts.md#failures-behave-like-exceptions-and-propagate-immediately).

Reply:
> If you've canceled the work, then you're not interested in any other results.

If the stream has already failed, I don't think emitting the pending value would make any difference.